lib/entice/logic/vitals.ex

```diff
@@ -1,54 +1,137 @@
+
+
+  def unregister(entity_id) do
+    Entity.remove_behaviour(entity_id, Vitals.AliveBehaviour)
+    Entity.remove_behaviour(entity_id, Vitals.DeadBehaviour)
+  end
+
+
+  @doc "Damage is dealt till death of the entity. (It then needs to be resurrected)"
+  def damage(entity, amount),
+  do: Coordination.notify(entity, {:vitals_entity_damage, amount})
 
 
-  def unregister(entity_id),
-  do: Entity.remove_behaviour(entity_id, Vitals.Behaviour)
+  @doc "Heals the entity until `max_health` is reached"
+  def heal(entity, amount),
+  do: Coordination.notify(entity, {:vitals_entity_heal, amount})
 
 
-  defmodule Behaviour do
+  @doc "Resurrect with percentage of health and energy. (Entity needs to be dead :P)"
+  def resurrect(entity, percent_health, percent_energy),
+  do: Coordination.notify(entity, {:vitals_entity_resurrect, percent_health, percent_energy})
+
+
+  defmodule AliveBehaviour do
     use Entice.Entity.Behaviour
     alias Entice.Logic.Vitals.Health
     alias Entice.Logic.Vitals.Energy
+    alias Entice.Logic.Vitals.Morale
+    alias Entice.Logic.Vitals.DeadBehaviour
     alias Entice.Logic.Player.Level
 
+    def init(
+        %Entity{attributes: %{Level => _, Morale => _}} = entity,
+        {:entity_resurrected, percent_health, percent_energy}) do
+      entity.id |> Coordination.notify_locally({
+        :entity_resurrected,
+        %{entity_id: entity.id, attributes: entity.attributes}})
+
+      %Health{max_health: max_health} = get_max_health(entity.attributes)
+      resurrected_health = round(max_health / 100 * percent_health)
+
+      %Energy{max_mana: max_mana} = get_max_energy(entity.attributes)
+      resurrected_mana = round(max_mana / 100 * percent_energy)
+
+      {:ok,
+        entity
+        |> put_attribute(%Health{health: resurrected_health, max_health: max_health})
+        |> put_attribute(%Energy{mana: resurrected_mana, max_mana: max_mana})}
+    end
+
+    def init(%Entity{attributes: %{Level => %Level{}}} = entity, _args) do
+      {:ok,
+        entity
+        |> put_attribute(%Morale{morale: 0})
+        |> attribute_transaction(
+           fn attrs ->
+             attrs |> Map.merge(%{
+               Health => get_max_health(attrs),
+               Energy => get_max_energy(attrs)})
+           end)}
+    end
 
-    def init(entity, _args) do
-      {:ok, entity |> put_attribute(get_max_health(entity))
-                   |> put_attribute(get_max_energy(entity))}
 
+    def handle_event(
+        {:vitals_entity_damage, amount},
+        %Entity{attributes: %{Health => %Health{health: health}}} = entity) do
+
+      new_health = health - amount
+      cond do
+        new_health >  0 ->
+          {:ok, entity |> update_attribute(Health, fn health -> %Health{health | health: new_health} end)}
+        new_health <= 0 ->
+          {:become, DeadBehaviour, :entity_died, entity |> update_attribute(Health, fn health -> %Health{health | health: 0} end)}
```

Comment:
Why not update health in death behaviour init. Will also solve problem of deaths not as a result of damage in which case we still want health = 0 (which exist but can't remember spell name at the moment).